services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -433,24 +436,77 @@ export class WorkflowService {
       projectIds,
     );
 
-    const totalWorkflowsCount = await this.workflowRuntimeDataRepository.count(
-      {
-        where: query.where,
-      },
-      projectIds,
+    const { workflowDefinitionIds, statuses, assigneeIds, includeUnassigned } = {
+      workflowDefinitionIds:
+        query.where.workflowDefinitionId &&
```

Comment:
It's hard to understand why this code is needed, also its repeating itself - I think extracting it into a function will make it more readable and concise